Documentation/async-disaster-recovery.md

```diff
@@ -0,0 +1,460 @@
+
+## Bootstrap Peers
+
+In order for the rbd-mirror daemon to discover its peer cluster, the
+ peer must be registered and a user account must be created. The following
+ steps enable bootstrapping peers to discover and authenticate to each other.
+
+For more details, refer to the official rbd mirror documentation on
+ [how to create a bootstrap peer](https://docs.ceph.com/en/latest/rbd/rbd-mirroring/#bootstrap-peers).
+
+## Create RBDMirror CRD
+
+Replication is handled by the rbd-mirror daemon. The rbd-mirror daemon
+ is responsible for pulling image updates from the remote, peer cluster,
+ and applying them to image within the local cluster.
+
+Creation of the rbd-mirror daemon(s) is done through the custom resource
+ definitions (CRDs).
+ For more information on how to set up Ceph RBDMirror CRD, refer to
+ [rook documentation](https://rook.io/docs/rook/master/ceph-rbd-mirror-crd.html).
+
+## Enable OMAP Generator and Volume Replication
+
+### OMap Generator
+
+Omap generator is a sidecar container that when deployed with
+ the CSI provisioner pod, generates the internal CSI omaps between
+ the PV and the RBD image. This is required as static PVs are
+ transferred across peer clusters in the DR use case, and hence
+ is needed to preserve PVC to storage mappings.
+
+### Volume Replication
+
+Volume Replication Operator is a kubernetes operator that provides common
+ and reusable APIs for storage disaster recovery.
+ It is based on [csi-addons/spec](https://github.com/csi-addons/spec)
+ specification and can be used by any storage provider.
+
+Volume Replication Operator follows controller pattern and provides
+ extended APIs for storage disaster recovery.
+ The extended APIs are provided via Custom Resource Definition (CRD).
+
+ Rook v1.6.0 comes with the new volume replication support, and
+ this can be enabled in the `rook-ceph-operator-config` configmap.
+
+>:bulb: For more information, please refer to the
+> [volume-replication-operator](https://github.com/csi-addons/volume-replication-operator).
+
+### Deploy `csi-omap-generator` and `volume-replication` sidecars
+
+To achieve RBD Mirroring, `csi-omap-generator` and `volume-replication`
+ containers need to be deployed in the RBD provisioner pods.
+Execute the following steps on each peer cluster to enable the
+ OMap generator and Volume Replication sidecars:
+
+* Edit the `rook-ceph-operator-config` configmap and add the
+ following configurations
+
+    ```bash
+    kubectl edit cm rook-ceph-operator-config  -nrook-ceph
+    ```
+
+    Add the following configuration if not present
+
+    ```yaml
+    apiVersion: v1
+    data:
+      CSI_ENABLE_OMAP_GENERATOR: "true"
+      CSI_ENABLE_VOLUME_REPLICATION: "true"
+    ```
+
+* After successful execution of the above steps, two new sidecars
+ should now come up in the CSI provisioner pod.
+* Repeat the steps on the peer cluster.
+
+## VolumeReplicationClass and VolumeReplication
```

Comment:
Everything needs to be phrased in terms of understanding of what the users needs to do. So can we rephrase the title to describe what is happening in the section, something like this?
```suggestion
## Enable Volume Replication
```